tasks.py

```diff
@@ -26,263 +26,174 @@
 if os.path.isfile(COMPOSE_OVERRIDE_FILE):
     COMPOSE_COMMAND += f' -f "{COMPOSE_OVERRIDE_FILE}"'
 
-NAUTOBOT_ROOT = "/opt/nautobot/"
+
+def docker_compose(context, command, **kwargs):
+    """Helper function for running a specific docker-compose command with all appropriate parameters and environment.
+
+    Args:
+        context (obj): Used to run specific commands
+        command (str): Command string to append to the "docker-compose ..." command, such as "build", "up", etc.
+        **kwargs: Passed through to the context.run() call.
+    """
+    print(f'Running docker-compose command "{command}"')
+    return context.run(f"{COMPOSE_COMMAND} {command}", env={"PYTHON_VER": PYTHON_VER}, **kwargs)
 
 
 # ------------------------------------------------------------------------------
 # BUILD
 # ------------------------------------------------------------------------------
 @task
-def build(context, python_ver=PYTHON_VER):
+def build(context, nocache=False, forcerm=False):
```

Comment:
Can we rename these to `no_cache` and `force_rm` respectively so that their CLI representations match those of Docker Compose? It ends up looking like:

```
$ invoke build -h
Usage: inv[oke] [--core-opts] build [--options] [other tasks here ...]

Docstring:
  Build all docker images.

  Args:
      context (obj): Used to run specific commands
      nocache (bool): Do not use cache when building the image
      forcerm (bool): Always remove intermediate containers

Options:
  -f, --force-rm
  -n, --no-cache
```

Reply:
I also think it would be good to start using the decorator to markup the help text for the options instead of providing it in the docstring:

```diff
diff --git a/tasks.py b/tasks.py
index bb7bc0bc..54c1a0ab 100644
--- a/tasks.py
+++ b/tasks.py
@@ -42,15 +42,14 @@ def docker_compose(context, command, **kwargs):
 # ------------------------------------------------------------------------------
 # BUILD
 # ------------------------------------------------------------------------------
-@task
-def build(context, nocache=False, forcerm=False):
-    """Build all docker images.
-
-    Args:
-        context (obj): Used to run specific commands
-        nocache (bool): Do not use cache when building the image
-        forcerm (bool): Always remove intermediate containers
-    """
+@task(
+    help={
+        "no_cache": "Do not use cache when building the image.",
+        "force_rm": "Always remove intermediate containers.",
+    },
+)
+def build(context, no_cache=False, force_rm=False):
+    """Build all docker images."""
     print("Building Nautobot .. ")
     command = f"build --build-arg PYTHON_VER={PYTHON_VER}"
     if nocache:
(END)
```

Which results in:

```
$ invoke build -h
Usage: inv[oke] [--core-opts] build [--options] [other tasks here ...]

Docstring:
  Build all docker images.

Options:
  -f, --force-rm   Always remove intermediate containers.
  -n, --no-cache   Do not use cache when building the image.
```

I don't think we should perpetuate Python functional docstrings for these user-facing tasks, since it is kind of confusing. For example, `context` is only ever used internally and therefore emitting that to to the CLI is weird.

Reply:
Great suggestions, thanks!